File: vue-app/src/classes/Treatment.ts
import { LocalIDFactory } from "@/helpers/LocalIDFactory";
import { IFile } from "@/interfaces/IFile";
import { IHoofCheck } from "@/interfaces/IHoofCheck";
import { ITreatment } from "@/interfaces/ITreatment";

export class Treatment implements ITreatment {
  localID?: string;
  note: string;
  noteForNextTreatment: string;
  horseId: number | undefined;
  horseName: string | undefined;
  date: Date;
  createdAt: Date;
  updatedAt: Date;
  fileKeysString: string;
  category: string;
  name: string;
  files: IFile[];
  treatmentIds: number[];
  hoofCheck: IHoofCheck | undefined;
  hoofCheckString: string | undefined;

  constructor() {
    this.localID = LocalIDFactory.createLocalID();
    this.note = "";
    this.noteForNextTreatment = "";
    this.date = new Date();
    this.createdAt = new Date();
    this.updatedAt = new Date();
    this.fileKeysString = "";
    this.category = "";
    this.name = "";
    this.files = [];
    this.treatmentIds = [];
  }
}

File: WebApi/Models/Treatment.cs
public class Treatment : IModel, IEnityWithFiles
{
    public int Id { get; set; }
    public string Note { get; set; } = string.Empty;
    public string NoteForNextTreatment { get; set; } = string.Empty;
    public int? HorseId { get; set; }
    public Horse? Horse { get; set; }
    public DateTime Date { get; set; } = DateTime.UtcNow;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public string Category { get; set; } = string.Empty;
    public string FileKeysString { get; set; } = string.Empty;
    public ICollection<File> Files { get; set; } = new List<File>();
    public string HoofCheckString { get; set; } = string.Empty;
}
